packages/plugins/typescript/react-apollo/src/visitor.ts

```diff
@@ -329,14 +331,19 @@ export class ReactApolloVisitor extends ClientSideBaseVisitor<ReactApolloRawPlug
     this.imports.add(this.getApolloReactCommonImport(true));
     this.imports.add(this.getApolloReactHooksImport(false));
 
+    const options = this.rawConfig.defaultBaseOptions
+      ? JSON.stringify(this.rawConfig.defaultBaseOptions)
+      : 'baseOptions';
     const hookFns = [
       `export function use${operationName}(baseOptions${
         hasRequiredVariables && operationType !== 'Mutation' ? '' : '?'
       }: ${this.getApolloReactHooksIdentifier()}.${operationType}HookOptions<${operationResultType}, ${operationVariablesTypes}>) {
+        const defaultBaseOptions = ${JSON.stringify(this.rawConfig.defaultBaseOptions)}
```

Comment:
Maybe we should merge `baseOptions` parameter from the function and defined `defaultBaseOption` like;
```ts
, { ...defaultBaseOptions, ...baseOptions}
```

Reply:
Perfect @ardatan ! There we add your recommendation and fix the tests

Reply:
Thank you! And also I think `const defaultBaseOptions` can be a global value because it will be duplicated in every single hook function in this way. Could you fix that as well? Then we can merge the PR :)

Reply:
And also could you fix the tests :) Thanks again!